feat(releases): migrate releases from gitlab to github

gaurav5430 opened the pull request:
Added code to transfer releases from gitlab to github with limited functionality (due to github api limitations).

spruce commented on src/githubHelper.ts:
```diff
@@ -160,6 +160,62 @@ export default class GithubHelper {
+   */
+   async getReleaseByTag(tag) {
+    try {
+      await utils.sleep(this.delayInMs);
```
Is the sleep here neccessary?

gaurav5430 replied:
it seems like this is a pattern we are following whenever we make the GitHub api call, to make sure that we do not exceed the quota. I have added this based on my understanding of the existing code

spruce replied:
For now we can leave it in. I just merged PRs relating to quatas / abuse limits. If it turns out it makes the migration slower than needed we will remove it later.

spruce commented on src/githubHelper.ts:
```diff
@@ -160,6 +160,62 @@ export default class GithubHelper {
+    name: string,
+    body: string) {
+    try {
+      await utils.sleep(this.delayInMs);
```
Is the sleep here neccessary?

gaurav5430 replied:
same as above

https://github.com/piceaTech/node-gitlab-2-github/pull/94#discussion_r763250387

spruce commented on src/index.ts:
```diff
@@ -149,6 +156,8 @@ async function migrate() {
         await transferMergeRequests();
       }
     }
+
+
```
Could you remove these newlines? :)

gaurav5430 replied:
would do

gaurav5430 replied:
removed

spruce commented on src/index.ts:
```diff
@@ -447,6 +456,63 @@ async function transferMergeRequests() {
+        console.error(err);
+      }
+    } else {
+      if (githubRelease) {
```
This if is always true because it is in the else branch of the inverse and can therefor be removed.

gaurav5430 replied:
would remove

gaurav5430 replied:
removed

spruce reviewed (changes requested):
Looks good sofar. But I found some things which I would like to be changed.
Also, please add a comment about the new feature to sample settings and the README.

Thanks 👍

gaurav5430 commented:
> Looks good sofar. But I found some things which I would like to be changed. Also, please add a comment about the new feature to sample settings and the README.
> 
> Thanks 👍

- added to README and sample settings
- replied to PR comments
- fixed whitespace
